src/index.ts

```diff
@@ -5,13 +5,16 @@ import 'core-js/es/array/includes';
 import 'promise-polyfill/src/polyfill';
 import 'fast-text-encoding';
 
+// @ts-ignore
 import Auth0Client from './Auth0Client';
 import * as ClientStorage from './storage';
-
-//this is necessary to export the type definitions used in this file
+import { Auth0ClientOptions } from './global';
 import './global';
```

Comment:
Is there any reason why it's left there? It looks like you can delete this line.

Reply:
This is to prevent breaking changes for people that are using the types when they are exposed globally by
```js
import createAuth0Client from '@auth0/auth0-spa-js';
```

Reply:
Actually after looking into this more, it looks like it actually is introducing breaking changes as the types are no longer globally scoped after being exported. I'm looking into a fix for this now

Reply:
@maxswa What is the best way to reproduce the failure with the type export that you're seeing?

Reply:
@maxswa Did you manage to look into this?

Reply:
If I import `import createAuth0Client from '@auth0/auth0-spa-js';` with my local version of the lib, I cannot use TypeScript types without explicitly importing them. I have not been able to find a solution that lets us export all of the types while also ambiently declaring them. Would this be an acceptable breaking change for this library? It would only effect TypeScript projects and the migration would be very straightforward.

Reply:
@maxswa I'm not sure that will at least make it any worse than it is now. Let me carve out some time on Monday to play with this but I don't think the types can be picked up ambiently at the moment anyway. Right now people have to import the types, but doing so by importing the specific type definition file rather than just importing from `@auth0/auth0-spa-js`.

Reply:
I'm playing with it now and the only problem I can see is that VS Code wants to auto-resolve the `Auth0Client` type to `@auth0/auth0-spa-js/dist/typings/Auth0Client`, but if I manually import it from the correct package, i.e.:

```js
import createAuth0Client, { Auth0Client } from "@auth0/auth0-spa-js";
```

It seems to work fine.